File: PythonProject/codeforces/swapSort.py
def sortWithSwaps(ar, n):
    swaps = []

    for i in range(n):
        min_index = i

        for j in range(i+1, n):
            if ar[j] < ar[min_index]:
                min_index = j
            
        if i != min_index:
            ar[i], ar[min_index] = ar[min_index], ar[i]
            swaps.append((i, min_index))
                
    print(len(swaps))
    for swap in swaps:
        print(swap[0], swap[1])
    

n = int(input())
array = list(map(int, input().split()))
sortWithSwaps(array, n)     

File: CppProject/codeforces/swapSort.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

void sortWithSwaps(vector<int>& ar, int n) {
    vector<pair<int, int>> swaps;

    for (int i = 0; i < n; ++i) {
        int min_index = i;

        for (int j = i + 1; j < n; ++j) {
            if (ar[j] < ar[min_index]) {
                min_index = j;
            }
        }

        if (i != min_index) {
            swap(ar[i], ar[min_index]);
            swaps.push_back({i, min_index});
        }
    }

    cout << swaps.size() << endl;
    for (const auto& swap : swaps) {
        cout << swap.first << " " << swap.second << endl;
    }
}

int main() {
    int n;
    cin >> n;
    vector<int> array(n);

    for (int i = 0; i < n; ++i) {
        cin >> array[i];
    }

    sortWithSwaps(array, n);

    return 0;
}
